src/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Sse41.cs

```diff
@@ -283,7 +283,7 @@ public static class Sse41
         /// __m128 _mm_insert_ps (__m128 a, __m128 b, const int imm8)
         ///   INSERTPS xmm, xmm/m32, imm8
         /// </summary>
-        public static Vector128<float> Insert(Vector128<float> value, float data, byte index) => Insert(value, data, index);
+        public static Vector128<float> Insert(Vector128<float> value, Vector128<float> data, byte index) => Insert(value, data, index);
```

Comment:
Corrected the SSE4.1 `Insert` API.

Reply:
Should we have both overloads or will we specially handle the `Sse.SetScalarVector128` and corresponding `index` masks to not zero the upper bits of `data`?

Reply:
Also, does the corresponding Avx API need to be updated as well?

Reply:
> Should we have both overloads 

Probably not, SSE4.1 `insertps` has really special semantics that inserts a value selected from the second vector argument into the first vector. But other SSE4.1 "insert" instructions direct insert a scalar value. https://msdn.microsoft.com/en-us/library/bb514071(v=vs.120).aspx

> Also, does the corresponding Avx API need to be updated as well?

No, AVX insert instructions are "normal" :)